### PR TITLE
Add benchmark to measure gc collection of a big chain of cycles

### PR DESCRIPTION
**Pull request by pablogsal**

### CODE REVIEW

**Comment by pablogsal** [comment]
More to come, I will create a PR per benchmark

File: pyperformance/data-files/benchmarks/bm_gc_collect/run_benchmark.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+CYCLES = 100
+LINKS = 20
+
+class Node:
+    def __init__(self):
+        self.next = None
+        self.prev = None
+    
+    def link_next(self, next):
+        self.next = next
```
Comment (markshannon) [inline pyperformance/data-files/benchmarks/bm_gc_collect/run_benchmark.py:13]: If you add `next.back = self` here you don't need the `link_back` method.

**Review by markshannon — COMMENTED** [review]
OOI do you have any numbers as to what fraction of the runtime this spends in GC?

**Comment by pablogsal** [comment]
> OOI do you have any numbers as to what fraction of the runtime this spends in GC?

By definition all of it, as the benchmark is only measuring `gc.collect()` call.

File: pyperformance/data-files/benchmarks/bm_gc_collect/run_benchmark.py (hunk 1 above)
Comment (pablogsal) [reply pyperformance/data-files/benchmarks/bm_gc_collect/run_benchmark.py:14]: Done

**Review by ericsnowcurrently — APPROVED** [review]
LGTM

**Pull request merged**